akka/src/main/scala/auth/directives/AuthDirectives.scala

```diff
@@ -107,6 +110,22 @@ trait AuthDirectives {
 
     respondWithHeader(Authorization(OAuth2BearerToken(token)))
   }
+
+  def authTokenCommand[T: Decoder, Y <: TokenCommand : Decoder]: Directive1[(T, Y)] = entity(as[Y]).flatMap { tokenHolder =>
+    Base64UnsafeCommandCrypto.decrypt[T](tokenHolder.token) match {
```

Comment:
- wtf Base64Unsafe? It should be a dependency
- AuthDirectives are those to be used by clients. They will never use token parsers. You can add these directly to action handler (and take command crypto service from auth services composition)